home/utils.py

```diff
@@ -217,86 +248,338 @@ def add_viber(row, page=None):
+            else:
+                row[0] = ""
+        elif row[1] == "x":
+            if row[5] == 1:
+                sub_1 += 1
+                row[1] = f"Sub {menu}.{sub_1}"
+                sub_2 = 0
+                sub_3 = 0
+                sub_4 = 0
+            else:
+                row[1] = ""
+        elif row[2] == "x":
+            if row[5] == 1:
+                sub_2 += 1
+                row[2] = f"Sub {menu}.{sub_1}.{sub_2}"
+                sub_3 = 0
+                sub_4 = 0
+            else:
+                row[2] = ""
+        elif row[3] == "x":
+            if row[5] == 1:
+                sub_3 += 1
+                row[3] = f"Sub {menu}.{sub_1}.{sub_2}.{sub_3}"
+                sub_4 = 0
+            else:
+                row[3] = ""
+        elif row[4] == "x":
+            if row[5] == 1:
+                sub_4 += 1
+                row[4] = f"Sub {menu}.{sub_1}.{sub_2}.{sub_3}.{sub_4}"
+            else:
+                row[4] = ""
+    return rows
+
+
+def get_rows(page: ContentPage) -> List[list]:
+    """Sets up row for each page including the side panel.
+    Each page is returned as a list of rows"""
+    actual_depth = (
+        page.depth - 3
+    )  # main menu is depth 3 (is this always true? root -> home -> main menu)
+    padding = [""] * 6
+    padding[actual_depth] = "x"
+    messenging_platform_messages = {
+        "whatsapp": get_messages(page.whatsapp_body),
+        "viber": get_messages(page.viber_body),
+        "messenger": get_messages(page.messenger_body),
+    }
 
     rows = []
-    for page in ContentPage.objects.all():
-        rows.append(get_rows(page))
-
-    write_rows(rows)
+    most_messages = max(
+        [len(messages) for messages in messenging_platform_messages.values()]
+        + [len(page.body)]
+    )
+    basic_row = padding + [
+        get_parent_page(page),
+        page.title,
+        page.subtitle,
+        str(page.body),
+        page.whatsapp_title,
+        messenging_platform_messages["whatsapp"][0]["msg"]
+        if len(messenging_platform_messages["whatsapp"]) == 1
+        else "",
+        page.messenger_title,
+        messenging_platform_messages["messenger"][0]["msg"]
+        if len(messenging_platform_messages["messenger"]) == 1
+        else "",
+        page.viber_title,
+        messenging_platform_messages["viber"][0]["msg"]
+        if len(messenging_platform_messages["viber"]) == 1
+        else "",
+        str(page.translation_key),
+        format_list_from_query_set(page.tags.all()),
+        format_list_from_query_set(page.quick_replies.all()),
+        format_list_from_query_set(page.triggers.all()),
+        str(page.locale),
+        get_next_prompts(messenging_platform_messages, 0),
+        get_image_link(messenging_platform_messages, 0),
+        get_doc_link(messenging_platform_messages, 0),
+        get_media_link(messenging_platform_messages, 0),
+        format_list_from_query_set(get_related_pages(page)),
+    ]
```

Comment:
Not for this PR, but maybe a change in the future, this section is the most confusing for me. There's nothing tying the arbitrary column positions with what they actually mean.

I'm wondering if using something like a dataclass here to hold all the data, might help better split the "getting the data" sections from the "formatting the data for presentation" sections.

eg.
```
@dataclass
class ExportRow:
    title: str
    subtitle: str
    body: str
```
Then once you've got all the rows, you can refer to things like `row.messenger_title`, which better describes what that actually is.